ios/Mappers.swift

```
@@ -1072,6 +1072,26 @@ class Mappers {
         }
     }
 
+    class func mapToSurchargeAmountOnly(from dict: [String: Any]?) throws -> Int? {
+        guard let dict = dict else { return nil }
+        
+        // iOS only supports 'amount' in the surcharge object on v4.5.0. If 'consent' is present, throw error.
+        if let consent = dict["consent"], !(consent is NSNull) {
+          throw NSError(
+            domain: "StripeTerminalReactNative",
+            code: ErrorCode.Code.invalidRequiredParameter.rawValue,
+            userInfo: [
+              NSLocalizedDescriptionKey: "iOS does not support 'consent' in surcharge"
+            ]
+          )
+        }
```

@lins-stripe could you, or someone more familiar with the surcharging feature, review this PR? should we throw an error or just ignore the consent for now on iOS? my concern with throwing the error is it could require platform specific checks for integrations that they'll have to remember to remove when this gets supported on iOS. But I guess there's also a concern of it not being passed through in the current state which is maybe worse 🤔

---

Yes, it's a valuable concern. I don't know how we dealt with this kind of situation before, so it's great to have your input!!

---

The implementation of this PR only accepts the client setting the amount. Everything should work fine. However, if the user sets the Consent, SDK throws an error.

Throwing Error has a concern that @bric-stripe mentioned. Regarding Ignoring Consent, it might confuse the user as to why setting this parameter doesn't affect anything. That's why I chose to throw an error. 

I don't have any preference for each solution, and it's nice to know how the team handles this generally.

---

[Summary] We will keep the current implementation and update the consent in future versions.
Thank you @bric-stripe & @lins-stripe 👍 

Updated, please review this again!